pg_wait_sampling.c

```diff
@@ -39,36 +34,42 @@
 
 PG_MODULE_MAGIC;
 
-void		_PG_init(void);
-
+/* Marker whether extension is setup in shared mode */
 static bool shmem_initialized = false;
 
+/* Global settings */
+int MaxProfileEntries = 5000;
+int HistoryBufferSize = 5000;
+int HistoryPeriod = 0;
+int ProfilePeriod = 10;
+bool WhetherProfilePid = true;
+bool WhetherProfileQueryId = true;
+
+/* Function declarations */
+void _PG_init(void);
+// TODO: add void _PG_fini(void);
+
 /* Hooks */
 static ExecutorEnd_hook_type	prev_ExecutorEnd = NULL;
 static planner_hook_type		planner_hook_next = NULL;
-
-/* Pointers to shared memory objects */
-shm_mq				   *pgws_collector_mq = NULL;
-uint64				   *pgws_proc_queryids = NULL;
-CollectorShmqHeader	   *pgws_collector_hdr = NULL;
-
-/* Receiver (backend) local shm_mq pointers and lock */
-static shm_mq *recv_mq = NULL;
-static shm_mq_handle *recv_mqh = NULL;
-static LOCKTAG queueTag;
-
 #if PG_VERSION_NUM >= 150000
-static shmem_request_hook_type prev_shmem_request_hook = NULL;
+static shmem_request_hook_type 	prev_shmem_request_hook = NULL;
 #endif
-static shmem_startup_hook_type prev_shmem_startup_hook = NULL;
-static PGPROC * search_proc(int backendPid);
+static shmem_startup_hook_type	prev_shmem_startup_hook = NULL;
 static PlannedStmt *pgws_planner_hook(Query *parse,
 #if PG_VERSION_NUM >= 130000
 		const char *query_string,
 #endif
 		int cursorOptions, ParamListInfo boundParams);
 static void pgws_ExecutorEnd(QueryDesc *queryDesc);
 
+/* Pointers to shared memory objects */
+pgwsQueryId *pgws_proc_queryids = NULL;
+HTAB		*pgws_profile_hash = NULL;
+LWLock		*pgws_profile_lock = NULL;
+History		*pgws_history_ring = NULL;
+LWLock		*pgws_history_lock = NULL;
```

Comment:
Maybe we should put these variables into a single struct pgws_shared_state or something like that?

Reply:
Do you mean put all these pointers into some structure? I think this doesn't bring smth even in code readability.